### DISCUSSION TITLE
Content Security Policy

### DISCUSSION THREAD

**Discussion by julianko** [directus/directus#15903] [Archive]
I'm trying to load a GLTF file in three.js inside a custom Module. Loading a GLTF automatically triggers loading of local blobs and I get errors like this:

`index.js:30903 Refused to connect to 'blob:http://localhost:8055/ed25b45a-5b6c-479a-82b4-dce63f2957a2' because it violates the document's Content Security Policy.`

I saw that there is a ASSETS_CONTENT_SECURITY_POLICY config variable which, I assume, will solve this problem. It says "See [helmet's documentation](https://helmetjs.github.io/) for more information.", but as far as I can tell, there is only documentation about how to use the JS api, so I'm a bit confused as to what I would write in an env variable.

**Comment by azrikahar** [comment]
> It says "See [helmet's documentation](https://helmetjs.github.io/) for more information.", but as far as I can tell, there is only documentation about how to use the JS api, so I'm a bit confused as to what I would write in an env variable.

Understandably there should be a more elaborate explanation on what to look for, or how to "translate" helmetjs' config to environment variables. Would you mind giving https://github.com/directus/directus/discussions/15865#discussioncomment-3786705 a read and let us know does it help?

**Reply by julianko** [reply]
It did help, thank you!

For anyone wondering, this is what solved the problem:
`CONTENT_SECURITY_POLICY_DIRECTIVES__CONNECT_SRC="array:'self',blob:,https://*"`

**Reply by rijkvanzanten** [reply]
cc @erondpowell

**Comment by reactone** [comment]
I get similar error when trying to access an mp3 file locally:
`Refused to load media from 'http://localhost:8055/assets/15c700d6-917c-4e36-9acb-cbe84a4f72fd' because it violates the following Content Security Policy directive: "default-src none". Note that 'media-src' was not explicitly set, so 'default-src' is used as a fallback.`

I am using Directus 9 and I have added the following to the .env file:
`CONTENT_SECURITY_POLICY_DIRECTIVES__CONNECT_SRC="media-src https://*"`

Now how do I restart directus for this new config to take effect? Using' npx directus start 'doesnt change anything and I dont want to install the new instance of Directus. @rijkvanzanten @julianko @azrikahar

**Reply by julianko** [reply]
Restarting Directus should change it, no need to reinstall.
Your env setting is not correctly formed though, try this:
`CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC="array:'self',blob:,https://*"`

**Reply by reactone** [reply]
Thanks for getting back to me, I've added the above to the .env file and run npx directus start, but it's still doesnt allow me to access the file directly in a new tab, showing the same error again. Any further advice? @julianko

**Comment by reactone** [comment]
> Restarting Directus should change it, no need to reinstall. Your env setting is not correctly formed though, try this: `CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC="array:'self',blob:,https://*"`

this worked: ASSETS_CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC"array:'self',blob:,https://*"
@julianko
